src/ExpressionGraph.jl

```diff
@@ -95,6 +95,18 @@ Base.zero(::Node) = Node(0)
 Base.one(::Type{Node}) = Node(1)
 Base.one(::Node) = Node(1)
 
+# These are essentially copied from Symbolics.jl:
+# https://github.com/JuliaSymbolics/Symbolics.jl/blob/e4c328103ece494eaaab2a265524a64bfbe43dbd/src/num.jl#L31-L34
+Base.eps(::Type{Node}) = Node(0)
+Base.typemin(::Type{Node}) = Node(-Inf)
+Base.typemax(::Type{Node}) = Node(Inf)
+Base.float(x::Node) = x
+
+# This one is needed because julia/base/float.jl only defines `isinf` for `Real`, but `Node
+# <: Number`.  (See https://github.com/brianguenter/FastDifferentiation.jl/issues/73)
+Base.isinf(x::Node) = !isnan(x) & !isfinite(x)
```

Comment:
I have questions about  isinf(x::Node). The other changes seem okay.

I'm guessing that isinf() is called in a conditional somewhere in Quaternions.jl, something like this:
```
function f(x)
  if isinf(x)
      return 1.0
  else
      return sin(x)
  end
end

The definition of isinf(x::Node) you are proposing will always return false so the FD expression tree will always be the expression evaluated in the non-infinite branch. 

In this case FD would generate derivative code that would return cos(x), even in the case when x happens to be inf. But f'(x) with x=inf is 0. So FD would return an incorrect derivative.

There's no easy way around this problem. I can add conditionals to FastDifferentiation, in the form of an ifelse function, but that won't solve the problem of conditionals in Julia code that use the built in Julia if...else statement. Which is basically everything that people might want to differentiate.

To get around this problem I'd have to analyze the user code, detect the if...else statement, and convert this somehow to an ifelse function call. Way beyond my level of Julia expertise.

Presumably the non-inf case is by far the most common, so defining isinf this way makes it possible to use FD with code that it otherwise wouldn't work with at the cost of potentially very mysterious derivative errors.